src/pages/WorkProgress.tsx

```diff
@@ -0,0 +1,234 @@
+    send({ type: "FIND", request: workProgressInput });
+    // Replace instead of push so user doesn't have to go through a load of old searches when going back
+    history.replace(`/?${stringify(workProgressInput)}`);
+  };
+
+  return (
+    <AppShell>
+      <AppShell.Header>
+        <AppShell.Title>STAN</AppShell.Title>
+      </AppShell.Header>
+      <AppShell.Main>
+        <div className="mx-auto">
+          <WorkProgressInput
+            initialValue={workProgressQueryInput}
+            onSubmitAction={handleSubmission}
+          />
+          <div className={"my-10"}>
+            {serverError && (
+              <Warning message="Search Error" error={serverError} />
+            )}
+            <div className="my-10">
+              {current.matches("searching") && (
+                <div className="flex flex-row justify-center">
+                  <LoadingSpinner />
+                </div>
+              )}
+            </div>
+            {current.matches("searched") ? (
+              searchResult && searchResult.entries.length > 0 ? (
+                <div>
+                  <div className="mt-6 mb-2 flex flex-row items-center justify-end">
+                    <p className="text-sm text-gray-700">
+                      Displaying{" "}
+                      <span className="font-medium">
+                        {" "}
+                        {searchResult.entries.length}{" "}
+                      </span>
+                      results
+                    </p>
+                  </div>
+                  <DataTable
+                    sortable
+                    defaultSort={[
+                      //Sort by Status and within status sort with WorkNumber in descending order
+                      {
+                        id: "status",
+                        desc: false,
+                      },
+                      {
+                        id: "workNumber",
+                        desc: true,
+                      },
+                    ]}
+                    columns={columns}
+                    data={searchResult.entries}
+                  />
+                </div>
+              ) : (
+                <Warning
+                  message={
+                    "There were no results for the given search. Please try again."
+                  }
+                />
+              )
+            ) : (
+              <div />
+            )}
+          </div>
+        </div>
+      </AppShell.Main>
+    </AppShell>
+  );
+}
+
+export default WorkProgress;
+
+/**
+ * Sort functionality for Date type
+ */
+const getDateSortType = (
+  rowADate: Date | undefined,
+  rowBDate: Date | undefined
+) => {
+  if (rowADate && rowBDate) {
+    return rowADate.getTime() - rowBDate.getTime();
+  }
+  if (rowADate) return 1;
+  return 0;
+};
+/**
+ * Sort functionality for Status. The status need to be sorted in the order "active", "completed", "paused", "failed"
```

Comment:
Was this one of the acceptance criteria? It may be overkill.

Reply:
Yes, This has came up later during the standup by Minal. The rows has to be ordered by status in the order -active,complete,paused,failed.) and within status SGP/R&D numbers need to be sorted in descending order. Please suggest if you have any better ways to improve performance.